### PR TITLE
Shape drawing doc

### PR DESCRIPTION
**Pull request by emmanuelle**
I've prepared the tutorials for shape drawing in preparation JS 1.54 / py 4.7. The CI will obviously fail until we upgrade the code to 1.54.

If you wish to test these tutorials locally, you should fetch this branch and run 
```
python setup.py updateplotlyjsdev --devbranch shape-drawing
```
to get the right plotly.js branch

### CODE REVIEW

File: doc/python/images.md [hunk 1]
```diff
@@ -305,5 +305,60 @@ fig.update_layout(
...
+
+It can be useful to add shapes to a layout image, for highlighting an object, drawing bounding boxes as part of a machine learning training set, or identifying seeds for a segmentation algorithm. 
+
+In order to enable shape drawing, you need to
+- define a dragmode corresponding to a drawing tool (`'drawline'`,`'drawopenpath'`, `'drawclosedpath'`, `'drawcircle'`, or `'drawrect'`)
+- add modebar buttons corresponding to the drawing tools you wish to use.
+
+The style of new shapes is specified by the `newshape` layout attribute. Shapes can be selected and modified after they have been drawn. More details and examples are given in the [tutorial on shapes](/python/shapes#drawing-shapes-on-cartesian-plots).
+
+Drawing or modifying a shape triggers a `relayout` event, which [can be captured by a callback inside a Dash application](https://dash.plotly.com/interactive-graphing).
```
Comment (archmoj) [inline doc/python/images.md:320]: Now it triggers `_guiRelayout` instead of `relayout`.

Comment (emmanuelle) [reply doc/python/images.md:320]: thanks I'll change this. For a Dash app, does `_guiRelayout` fall in the realm of `relayoutData` events? I tried with the bundle from yesterday afternoon and my Dash apps still work as expected with callbacks listening to `relayoutData`.

Comment (nicolaskruchten) [reply doc/python/images.md:320]: I think we don't need to tell Python users the name of the actual underlying event, just how to access it from widgets and Dash.

Comment (archmoj) [reply doc/python/images.md:320]: OK. Resolving...

File: doc/python/shapes.md [hunk 2]
```diff
@@ -716,5 +716,89 @@ fig.update_layout(
...
+            text=text,
+            xref="paper",
+            yref="paper",
+            showarrow=False,
+            font_size=20
+)
+# shape defined programatically
+fig.add_shape(line_color='yellow',
+              fillcolor='turquoise',
+              opacity=0.8,
```
Comment (archmoj) [inline doc/python/shapes.md:784]: You may consider using `opacity <= 0.5` here & below if you want to allow users to start drawing new shapes from inside of another shapes.

File: doc/python/images.md [hunk 3]
```diff
@@ -305,5 +305,60 @@ fig.update_layout(
 fig.show(config={'doubleClick': 'reset'})
 ```
 
+### Annotating layout image with shapes
+
+_introduced in plotly 4.7_
```
Comment (nicolaskruchten) [inline doc/python/images.md:310]: Yeah, this is probably a good practice :)

File: doc/python/images.md [hunk 4]
```diff
@@ -305,5 +305,60 @@ fig.update_layout(
...
+
+It can be useful to add shapes to a layout image, for highlighting an object, drawing bounding boxes as part of a machine learning training set, or identifying seeds for a segmentation algorithm. 
+
+In order to enable shape drawing, you need to
+- define a dragmode corresponding to a drawing tool (`'drawline'`,`'drawopenpath'`, `'drawclosedpath'`, `'drawcircle'`, or `'drawrect'`)
+- add modebar buttons corresponding to the drawing tools you wish to use.
```
Comment (nicolaskruchten) [inline doc/python/images.md:316]: link to the config doc here?

File: doc/python/images.md [hunk 5]
```diff
@@ -305,5 +305,60 @@ fig.update_layout(
...
+
+```python
+import plotly.graph_objects as go
+fig = go.Figure()
+# Add image
+img_width = 1600
+img_height = 900
+scale_factor = 0.5
+fig.add_layout_image(
+    dict(
```
Comment (nicolaskruchten) [inline doc/python/images.md:330]: the `dict` here is redundant I'm pretty sure, as `add_layout_image` accepts these values as kwargs

File: doc/python/shapes.md [hunk 6]
```diff
@@ -716,5 +716,89 @@ fig.update_layout(
 fig.show()
 ```
 
+### Drawing shapes on Cartesian plots
+
+_introduced in plotly 4.7_
+
+You can create layout shapes programatically, but you can also draw shapes manually by setting the `dragmode` to one of the shape-drawing modes: `'drawline'`,`'drawopenpath'`, `'drawclosedpath'`, `'drawcircle'`, or `'drawrect'`. If you need to switch between different shape-drawing or other dragmodes (panning, selecting, etc.), modebar buttons can be added in the `config` to select the dragmode. If you switch to a different dragmode such as pan or zoom, you will need to select the drawing tool in the modebar to go back to shape drawing.
```
Comment (nicolaskruchten) [inline doc/python/shapes.md:723]: add link to config doc?

File: doc/python/shapes.md (hunk 2 above)
Comment (emmanuelle) [reply doc/python/shapes.md:784]: thanks !

File: doc/python/shapes.md (hunk 6 above)
Comment (emmanuelle) [reply doc/python/shapes.md]: done!

File: doc/python/images.md (hunk 5 above)
Comment (emmanuelle) [reply doc/python/images.md]: yes!

File: doc/python/images.md (hunk 4 above)
Comment (emmanuelle) [reply doc/python/images.md]: done!

**Comment by emmanuelle** [comment]
The CI is now green!

**Comment by nicolaskruchten** [comment]
:dancer: !

**Pull request merged**